test/blackbox-tests/test-cases/pkg/pkg-enabled.t

```diff
@@ -39,3 +39,32 @@ When the default lockdir is present pkg is enabled:
 When a non-default lockdir is present, pkg is still enabled:
   $ dune pkg lock dune.other.lock > /dev/null 2> /dev/null
   $ dune pkg enabled
+
+Remove the other lock dir and make sure the status didn't latch to be always
+enabled.
+
+  $ rm -r dune.other.lock
+  $ dune pkg enabled
+  [1]
+
+Manually enable package management in the workspace
+
+  $ cat > dune-workspace <<EOF
+  > (lang dune 3.20)
+  > (pkg enabled)
+  > EOF
+  $ dune pkg enabled && echo "Yes, it is enabled"
+  Yes, it is enabled
+
+  $ cat > dune-workspace <<EOF
+  > (lang dune 3.20)
+  > EOF
+  $ dune pkg enabled || echo "Package management disabled"
+  Package management disabled
+
+  $ cat > config <<EOF
+  > (lang dune 3.20)
+  > (pkg enabled)
+  > EOF
+  $ dune pkg enabled --config-file=config && echo "Yes, it is enabled"
+  Yes, it is enabled
```

Comment:
Discussed offline, but a test to check the overriding behavior between user config and workspace config seems like a good idea.